### PR TITLE
Onboarding logic for nonauth and closed campaigns

### LINKED ISSUE #6662: Display onboarding on a campaign signup
#### User Story

As a user, after I signup for a campaign I would like to see onboarding message box 
#### Additional Information (optional)

This should happen the first time and only the first time the authenticated user sees the action page.
This should appear on every single signup the user makes

### LINKED ISSUE #6729: Disable the Onboarding component on Closed Campaigns
@jessleenyc @lkpttn  I'm assuming we _don't_ want to show the Onboarding stuff on closed campaigns? I believe currently it will display on those campaigns. While considering this, are there other campaign types we don't want it to show on? Grouped campaigns (are those still a thing?)

cc: @angaither

### PR DESCRIPTION
**Pull request by weerd**
#### What's this PR do?

This PR moves the additional and checks for adding JS scripts into a dosomething_campaign theme helper function `dosomething_helpers_add_js()`, which is called from the `dosomething_campaign_preprocess_node()` function in the **dosomething_campaign.theme.inc** file to remove the need to do those checks in a later general campaign node preprocess function. 

It also adds logic to keep the onboarding component from showing up for authenticated users on campaign pitch pages, closed campaigns, sms game campaigns, and on any campaign related pages for anonymous users.

It also implements the use of `localStorage` to house a key/value that will determine if the user has already seen the Onboarding feature on a specified campaign; if so, then it does not show it again. This provides a quick means to enable this functionality and only show Onboarding on first signup, without having to commit to creating a new database table and data for this prototype.
#### How should this be reviewed?

Try hitting a closed campaign while logged in and while logged out. Do you see the onboarding component pop up? No? Great!
#### Relevant tickets

Fixes #6729
Fixes #6662
#### Checklist
- [ ] Tested on staging.

---

@angaither @deadlybutter @DFurnes 

cc: @jessleenyc

### CODE REVIEW

**Comment by angaither** [comment]
@weerd this still needs to be happening after a signup event, so the checks for this should have happened before this preprocess_node function

File: lib/modules/dosomething/dosomething_helpers/dosomething_helpers.theme.inc [hunk 1]
```diff
@@ -269,3 +269,45 @@ function dosomething_helpers_add_inline_css(&$vars) {
...
+      ],
+      'dsCampaign' => $current_campaign,
+    ],
+    'setting'
+  );
+
+  // @Temporary: add lab.js with Onboarding experiment.
+  if ($current_user && variable_get('dosomething_campaign_enable_onboarding', FALSE)) {
+    if ($current_campaign->status !== 'closed' && $current_campaign->type !== 'sms_game' && !dosomething_campaign_is_pitch_page($vars['node'])) {
+      drupal_add_js(PARANEUE_PATH . '/dist/lab.js', [
```
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_helpers/dosomething_helpers.theme.inc]: 🔬

**Comment by itsjoekent** [comment]
local storage looks good :D 🍨

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc [hunk 2]
```diff
@@ -29,6 +29,9 @@ function dosomething_campaign_preprocess_node(&$vars) {
   // Add inline css based on vars.
   dosomething_helpers_add_inline_css($vars);
 
+  // Add internal js script based on vars.
+  dosomething_helpers_add_js($vars);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc]: move adding onboaring JS down below the checks for pitch page/sms and closed campaigns

File: lib/modules/dosomething/dosomething_helpers/dosomething_helpers.theme.inc [hunk 3]
```diff
@@ -269,3 +269,45 @@ function dosomething_helpers_add_inline_css(&$vars) {
...
+        'activity' => dosomething_helpers_get_current_user_activity($vars['user']),
+      ],
+      'dsCampaign' => $current_campaign,
+    ],
+    'setting'
+  );
+
+  // @Temporary: add lab.js with Onboarding experiment.
+  if ($current_user && variable_get('dosomething_campaign_enable_onboarding', FALSE)) {
+    if ($current_campaign->status !== 'closed' && $current_campaign->type !== 'sms_game' && !dosomething_campaign_is_pitch_page($vars['node'])) {
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_helpers/dosomething_helpers.theme.inc]: then these checks don't need to happen before adding onboarding js

**Comment by weerd** [comment]
@angaither works! Thanks for the recommendations.

Updated to add the Onboarding script after the already present campaign page checks in `dosomething_campaign_preprocess_node()`.

The `dsCampaign` and `dsUser` objects are added to all campaign pages, since it'll be useful for future React components or other JS things. The Onboarding script is only added to action pages.

**Comment by angaither** [comment]
🚢

**Pull request merged**